src/Reflection/Php/PhpFunctionReflection.php

```diff
@@ -167,41 +165,40 @@ private function isVariadic(): bool
 	}
 
 	/**
-	 * @param Node[] $nodes
+	 * @param Node[]|scalar[]|Node $node
 	 */
-	private function callsFuncGetArgs(array $nodes): bool
+	private function callsFuncGetArgs(array|Node $node): TrinaryLogic
 	{
-		foreach ($nodes as $node) {
+		$result = TrinaryLogic::createMaybe();
+
+		if ($node instanceof Node) {
 			if ($node instanceof Function_) {
 				$functionName = (string) $node->namespacedName;
 
 				if ($functionName === $this->reflection->getName()) {
-					return $this->functionCallStatementFinder->findFunctionCallInStatements(ParametersAcceptor::VARIADIC_FUNCTIONS, $node->getStmts()) !== null;
+					return TrinaryLogic::createFromBoolean($this->functionCallStatementFinder->findFunctionCallInStatements(ParametersAcceptor::VARIADIC_FUNCTIONS, $node->getStmts()) !== null);
```

Comment:
I think I'm missing looking at `Param::$variadic` here. Now it looks like it treats `function foo($bar) { }` and `function foo(...$bar) { }` the same and passes the function as variadic in either case. I think it should be more strict.

Reply:
native variadic is already [checked in the outer scope](https://github.com/phpstan/phpstan-src/pull/3370/files#diff-3ac4e5d4a520618a490963b70e3fcd428d56aea61683641020393167fff2ca5dR140-R141).

for native variadic we don't call into `callsFuncGetArgs` at all

Reply:
Please reproduce this situation and fix it, thanks.

Reply:
I was not able to reproduce it initially because of ghe caching involved. 

After temporarily using different cache keys locally the case was showing up :-)